processor/src/host/advice/mod.rs

```diff
@@ -45,6 +47,29 @@ pub struct AdviceProvider {
 }
 
 impl AdviceProvider {
+    /// Apply the mutations given in order to the `AdviceProvider`.
+    pub fn apply_mutations(
```

Comment:
We may be able to get rid of some of the mutation functions now. For example, I'm not sure `push_stack` and `push_stack_word` are needed any more. But we can do this also later.